packages/compartment-mapper/src/infer-exports.js

```diff
@@ -57,6 +57,7 @@ function* interpretBrowserField(name, browser, main = 'index.js') {
 function* interpretExports(name, exports, tags) {
   if (isArray(exports)) {
     for (const section of exports) {
+      /** @type {Array} */
```

Comment:
what happens without this? is TS not picking up the `yields` return type?

Reply:
No, the issue seems to be with `[...val]` (and `Array.from(val)`, for that matter) not being recognized as an array.

Reply:
That was because `val` was `any`, because `interpretExports` was returning `any`, because https://github.com/Microsoft/TypeScript/issues/23857

I've got a fix in https://github.com/endojs/endo/tree/ta/gibson-2023-02-type-aware-linting